core/src/main/java/hudson/model/ListView.java

```diff
@@ -412,13 +413,15 @@ public HttpResponse doRemoveJobFromView(@QueryParameter String name) throws IOEx
         if(name==null)
             throw new Failure("Query parameter 'name' is required");
 
-        TopLevelItem item = resolveName(name);
-        if (item==null)
-            throw new Failure("Query parameter 'name' does not correspond to a known and readable item");
-
-        if (remove(item))
-            owner.save();
-
+        synchronized (this) {
```

Comment:
@liketic's PR didn't contain this sync block, which should prevent race conditions with other methods like `submit()`